lib/active_merchant/billing/gateways/wompi.rb

```diff
@@ -0,0 +1,249 @@
+
+        add_invoice(post, money, options)
+        add_payment(post, payment, options)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+
+        commit(:post, '/transactions', post)
+      end
+
+      # Lookup transaction using reference or using transaction id path parans
+      #
+      def query_transaction(reference, options={})
+        if options[:path_params] && reference.present?
+          commit(:get, "/transactions/#{reference}")
+        else
+          commit(:get, "/transactions?reference=#{reference}")
+        end
+      end
+
+      def pse_financial_institutions
+        commit(:get, "/pse/financial_institutions")
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(/(Authorization: Bearer )([^\s])+/, '\1[FILTERED]').
+          gsub(/(\\?\\?\\?"number\\?\\?\\?":\\?\\?\\?")\d+/, '\1[FILTERED]').
+          gsub(/(\\?\\?\\?"cvc\\?\\?\\?":\\?\\?\\?"?)\d+/, '\1[FILTERED]')
+      end
+
+      private
+
+      def add_customer_data(post, options)
+        post[:customer_data] = {}
+        post[:customer_data][:phone_number] = options.dig(:customer, :mob_phone)
+        post[:customer_data][:full_name] = options.dig(:customer, :full_name)
+        post[:customer_email] = options.dig(:customer, :email)
+        post[:redirect_url] = options[:redirect_url]
+      end
+
+      def add_address(post, creditcard, options)
+        post[:shipping_address] = add_address_data(options)
+      end
+
+      def add_address_data(options)
+        address = {}
+
+        if options[:address].present?
+          address[:address_line_1] = options[:address][:address1]
+          address[:address_line_2] = options[:address][:address2]
+          address[:city] = options[:address][:city]
+          address[:region] = options[:address][:state]
+          address[:name] = options[:customer]&.[](:full_name)
+          address[:phone_number] = options[:customer]&.[](:mob_phone)
+          address[:postal_code] = options[:address][:postal_code]
+          address[:country] = options[:address][:country]
+        end
+
+        address
+      end
+
+      def add_invoice(post, money, options)
+        post[:amount_in_cents] = amount(money).to_i
+        post[:currency] = (options[:currency] || currency(money))
+        post[:reference] = options[:reference]
+      end
+
+      def add_payment(post, payment, options)
+        post[:payment_method] = {}
+        post[:payment_method][:type] = "CARD"
+        post[:payment_method][:token] = options[:token]
+        post[:payment_method][:installments] = options[:installments] || 1
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def headers(action)
+        key = reference_in_params?(action) ? :private_key : :public_key
+        token = @options.send(:[], key)
+
+        raise 'Missing Bearer token' if token.nil?
```

Comment:
```suggestion
        token = @options.fetch(key)
```

Is shorter and the error message says exactly which key was expected. However I just noticed that you check for options above with `requires!` so my original comment was wrong, you don't necessarily need to check for presence here. Then again it can't hurt to make the expectation clear here.